File: LeetCode/20_valid_parentheses/isValid.cpp
class Solution {
public:
    bool isValid(string s) {
        std::stack<char> stack;
        for (int i = 0; i < s.length(); i++) {
            char c = s[i];
            if (c == '(' || c == '{' || c == '[')   stack.push(c);
            else if (stack.empty()) return false;
            else {
                char top = stack.top();
                if (stack.empty())  return false;
                if (c == ')' && top != '(') return false;
                if (c == '}' && top != '{') return false;
                if (c == ']' && top != '[') return false;
                stack.pop();
            }
        }
        return stack.empty();
    }
};

 
Add to List
20. Valid Parentheses

File: LeetCode/20_valid_parentheses/isValid.py
class Solution(object):
    def isValid(self, s):
        stack = []
        dict = {"}":"{", "]":"[", ")":"("}
        for c in s:
            if c in dict.values():
                stack.append(c)
            elif c in dict.keys():
                if (stack == [] or dict[c] != stack.pop()):
                    return False
            else:
                return False
        return (stack == [])

